typescript/looker/rtl/extensionSession.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2019 Looker Data Sciences, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+import { IApiSettings } from './apiSettings'
+import { ITransport, IRequestProps } from './transport'
+import { IAuthSession } from './authSession'
+
+export class ExtensionSession implements IAuthSession {
+  sudoId: string = ''
+  transport: ITransport
+
+  constructor(public settings: IApiSettings, transport: ITransport) {
```

Comment:
minor typescript syntax sugar you may already know about, you could do

```typescript
constructor(public settings: IApiSettings, public transport: ITransport)
```
instead of explicitly declaring `transport` above. 

That also reminds me I want to find out if the explicit setters for `this.settings` and `this.transport` are even required if the property scope is declared in the constructor signature.

Reply:
I just did a straight copy from  the extension SDK and it had some "fixes" to get it to compile after the last core SDK change. Now that the class is in the coreSDK I can see it needs a little more adjustment. Will push that change soon.

Reply:
Fixed as per your suggestion